inbox/api/views.py

```diff
@@ -0,0 +1,70 @@
+# =================================================================================================
+#                                  All Rights Reserved.
+# =================================================================================================
+# File description:
+#       In other frameworks you may also find conceptually similar implementations named
+#       something like 'Resources' or 'Controllers'.
+#
+#       Ref: https://www.django-rest-framework.org/api-guide/viewsets/
+#
+# =================================================================================================
+#    Date      Name                    Description of Change
+# 13-Mar-2022  Wayne Shih              Initial create
+# $HISTORY$
+# =================================================================================================
+
+
+from notifications.models import Notification
+from rest_framework import viewsets, status
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from inbox.api.serializers import NotificationSerializer, DefaultNotificationSerializer
+
+
+class NotificationViewSet(viewsets.GenericViewSet):
+    queryset = Notification.objects.all()
+    serializer_class = DefaultNotificationSerializer
+    permission_classes = (IsAuthenticated,)
+    # <Wayne Shih> 15-Mar-2022
+    # filterset_fields is a shortcut for filterset_class
+    # https://django-filter.readthedocs.io/en/latest/guide/rest_framework.html#adding-a-filterset-with-filterset-class
+    # https://django-filter.readthedocs.io/en/latest/guide/rest_framework.html#using-the-filterset-fields-shortcut
+    filterset_fields = ('unread',)
+
+    # <Wayne Shih> 13-Mar-2022
+    # URL:
+    # - GET /api/notifications/
+    def list(self, request: Request):
+        queryset = self.get_queryset()
+        notifications = self.filter_queryset(queryset)\
+            .filter(recipient=request.user)\
+            .order_by('-timestamp')\
+            .prefetch_related('actor', 'target')
+        serializer = NotificationSerializer(notifications, many=True, context={'request': request})
+        return Response({
+            'notifications': serializer.data,
+        }, status=status.HTTP_200_OK)
```

Comment:
you might also consider to use ListModelMixin

Reply:
Not sure if ListModelMixin supports prefetch_related?

Reply:
In this case, you could try to make 
filter(), order_by(), and prefetch_related() all in the get_queryset()